source/ranges.tex

```diff
@@ -9341,6 +9357,550 @@
+\begin{codeblock}
+namespace std::ranges {
+  template<@\libconcept{view}@ V>
+    requires @\exposconcept{range-with-movable-references}@<V>
+  template<bool Const>
+  class enumerate_view<V>::@\exposidnc{sentinel}@ {                           // \expos
+    using @\exposidnc{Base}@ = @\exposidnc{maybe-const}@<Const, V>;                         // \expos
+    sentinel_t<@\exposidnc{Base}@> @\exposidnc{end_}@ = sentinel_t<@\exposidnc{Base}@>();                 // \expos
+    constexpr explicit @\exposid{sentinel}@(sentinel_t<@\exposid{Base}@> end);
+
+  public:
+    @\exposid{sentinel}@() = default;
+    constexpr @\exposid{sentinel}@(@\exposid{sentinel}@<!Const> other)
+      requires Const && @\libconcept{convertible_to}@<sentinel_t<V>, sentinel_t<@\exposid{Base}@>>;
+
+    constexpr sentinel_t<@\exposid{Base}@> base() const;
+
+    template<bool OtherConst>
+      requires @\libconcept{sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+    friend constexpr bool operator==(const @\exposid{iterator}@<OtherConst>& x, const @\exposid{sentinel}@& y);
+
+    template<bool OtherConst>
+      requires @\libconcept{sized_sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+    friend constexpr range_difference_t<@\exposid{maybe-const}@<OtherConst, V>>
+      operator-(const @\exposid{iterator}@<OtherConst>& x, const @\exposid{sentinel}@& y);
+
+    template<bool OtherConst>
+      requires @\libconcept{sized_sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
+    friend constexpr range_difference_t<@\exposid{maybe-const}@<OtherConst, V>>
+      operator-(const @\exposid{sentinel}@& x, const @\exposid{iterator}@<OtherConst>& y);
+  };
+}
+\end{codeblock}
+
+\indexlibraryctor{enumerate_view::\exposid{sentinel}}%
+\begin{itemdecl}
+constexpr explicit @\exposid{sentinel}@(sentinel_t<@\exposid{Base}@> end);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{end_} with \tcode{std::move(end)}.
+\end{itemdescr}
+
+\indexlibraryctor{enumerate_view::\exposid{sentinel}}%
+\begin{itemdecl}
+constexpr @\exposid{sentinel}@(@\exposid{sentinel}@<!Const> other)
+  requires Const && @\libconcept{convertible_to}@<sentinel_t<V>, sentinel_t<@\exposid{Base}@>>;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Initializes \exposid{end_} with \tcode{std::move(other.\exposid{end_})}.
+\end{itemdescr}
+
+\indexlibrarymember{base}{enumerate_view::\exposid{sentinel}}%
+\begin{itemdecl}
+constexpr sentinel_t<@\exposid{Base}@> base() const;
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return \exposid{end_};}
+\end{itemdescr}
+
+\indexlibrarymember{operator==}{enumerate_view::\exposid{sentinel}}%
+\begin{itemdecl}
+template<bool OtherConst>
+  requires sentinel_for<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
```

Comment:
```suggestion
  requires @\libconcept{sentinel_for}@<sentinel_t<@\exposid{Base}@>, iterator_t<@\exposid{maybe-const}@<OtherConst, V>>>
```
Thankfully, this is a rare occurrence now.

Reply:
Good catch, thanks! Our build checks should have caught this.

Reply:
Oh, right! I think I remember that was added. @jensmaurer the check failed here and below.